authentication/authentication.go

```diff
@@ -0,0 +1,153 @@
+package authentication
+
+import (
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
+	"io/ioutil"
+)
+
+// wraps raw data in JWT and JWE
+func GetJwe(rawClaims interface{}, signingKeyPath string, encryptionKeyPath string) (string, *TokenError) {
+
+	privateKeyResult, keyErr := loadSigningKey(signingKeyPath)
+
+	if keyErr != nil {
+		return "", &TokenError{Desc: "Error loading signing key", From: keyErr}
+	}
+
+	publicKeyResult, keyErr := loadEncryptionKey(encryptionKeyPath)
+	if keyErr != nil {
+		return "", &TokenError{Desc: "Error loading encryption key", From: keyErr}
+	}
+
+	opts := jose.SignerOptions{}
+	opts.WithType("JWT")
+	opts.WithHeader("kid", privateKeyResult.kid)
+
+	signer, err := jose.NewSigner(jose.SigningKey{Algorithm: jose.RS256, Key: privateKeyResult.key}, &opts)
+	if err != nil {
+		return "", &TokenError{Desc: "Error creating JWT signer", From: err}
+	}
+
+	encrypter, err := jose.NewEncrypter(
+		jose.A256GCM,
+		jose.Recipient{Algorithm: jose.RSA_OAEP, Key: publicKeyResult.key, KeyID: publicKeyResult.kid},
+		(&jose.EncrypterOptions{}).WithType("JWT").WithContentType("JWT"))
+
+	if err != nil {
+		return "", &TokenError{Desc: "Error creating JWT signer", From: err}
+	}
+
+	token, err := jwt.SignedAndEncrypted(signer, encrypter).Claims(rawClaims).CompactSerialize()
+
```

Comment:
Should avoid blank lines between the method call and the error handler

Reply:
Done